1-Draft/RFC00XX-Update-Error-View.md

```diff
@@ -0,0 +1,226 @@
+
+- `Get-Error` will provide the following:
+
+    - Display the newest Error ($Error[0]) – default behavior
+    - Accept Pipeline input – support $error[1] |
+    - Option for the Newest X errors in the session (This will be aliased with Last)
+
+- `Get-Error` syntax
+
+```powershell
+Get-Error  [-InputObject <psobject>] [-Newest <Int32>] [-All] [<CommonParameters>]
+```
+
+First parameter set
+
+- Newest
+
+    + Datatype: int32
+    + specifies one or more of the newest errors to display
+    + Not required
+
+__Example 1__
+Error occurs in Interactive mode. Cmdlet displays details of the last error displayed
+
+```powershell
+PS C:\> Get-Childitem -Path c:\notreal
+Get-ChildItem: Cannot find path ‘C:\notreal’ because it does not exist
+
+PS C:\test> Get-Error
+
+**** Detailed message here ****
+```
+
+__Example 2__
+Error occurs in script, shows error from view 'Analytic', and then is piped
+from $error array to 'Get-Error' to display more details.
+
+```powershell
+PS C:\> .\MyScript.ps1
+Get-ChildItem: in /Users/Username/GitHub/Errorview/script.test.ps1
+    |
+15  | Get-ChildItem -Path c:\notreal
+    |                     ^^^ Cannot find path 'C:\notreal' because it does not exist.
+    * Help: Additional help information provided here
+
+PS C:\> $error[0] | Get-Error
+
+**** Detailed message here ****
+```
+
+__Example 3__
+Display detailed error information for the most recent 3 errors.
+
+```powershell
+PS C:\> Get-Error -Newest 3
+
+**** Detailed message here ****
+**** Detailed message here ****
+**** Detailed message here ****
+```
+
+__Example 4__
+Maintain the ErrorRecord object for additional pipeline operations
+
+```PowerShell
+PS C:\> Get-Error -Newest 3 | Select-String -Pattern 'MyFile.txt'
+
+**** Detailed message here ****
+```
+
+## Alternate Proposals and Considerations
+
+__Alternative/additional view customization__
+
+It is conceivable in the future to add extensibility for module builders,
+that they could supply their own diagnostic script for specific error customization.
+
+__Alternative single-line display__
+
+```powershell
+<CommandName>:<Exception Message>
+get-item: Cannot find path ‘C:\blah’ because it does not exist:
+
+<CommandName>:<Exception Message>: <Position>
+get-item: Cannot find path ‘C:\blah’ because it does not exist: At line:1 char:1
+
+<CommandName>:<Alias>:<Exception Message>: <Position>
+get-item: gi: Cannot find path ‘C:\blah’ because it does not exist: At line:1 char:1
+
+ERROR:<Exception Message>
+ERROR: Cannot find path ‘C:\blah’ because it does not exist
+
+ERROR:<Exception Message>: <Position>
+ERROR: Cannot find path ‘C:\blah’ because it does not exist: At line:1 char:1
+
+ERROR:<CommandName>:<Alias>:<Exception Message>: <Position>
+ERROR: get-item: gi: Cannot find path ‘C:\blah’ because it does not exist: At line:1 char:1
+```
+
+__Alternative color for all errors__
+
+1. We could change the default error message color from the current RED foreground and BLACK background to ?.
```

Comment:
whatever we do, it needs to be configurable for accessibility reasons

Reply:
Agreed. Currently, the only color change has been with new Accent color and can be changed for preference under $host.PrivateData.ErrorAccentColor.

Reply:
(I state up front that this doesn't apply to me so I would be a poor test case.) Have you tested the color changes against someone with male-pattern color blindness?